runtime/lua/vim/iter.lua

```diff
@@ -179,6 +208,55 @@ function ListIter.filter(self, f)
   return self
 end
 
+--- Flattens a |list-iterator|, un-nesting nested values up to the given {depth}.
+--- Errors if it attempts to flatten a dict-like value.
+---
+--- Examples:
+---
+--- ```lua
+--- vim.iter({ 1, { 2 }, { { 3 } } }):flatten():totable()
+--- -- { 1, 2, { 3 } }
+---
+--- vim.iter({1, { { a = 2 } }, { 3 } }):flatten():totable()
+--- -- { 1, { a = 2 }, 3 }
+---
+--- vim.iter({ 1, { { a = 2 } }, { 3 } }):flatten(math.huge):totable()
+--- -- error: attempt to flatten a dict-like table
+--- ```
+---@param depth? number Depth to which |list-iterator| should be flattened
```

Comment:
This should fix the doc issues

```suggestion
--- ```
---
---@param depth? number Depth to which |list-iterator| should be flattened
```

Reply:
thank you!